tests/testSpherePoint.py

```diff
@@ -0,0 +1,751 @@
+            for element in point.getVector():
+                self.assertTrue(math.isnan(element))
+
+    def testTicket1761(self):
+        """Regression test for Ticket 1761.
+
+        Checks for math errors caused by unnormalized vectors.
+        """
+        refPoint = SpherePoint(afwGeom.Point3D(0, 1, 0))
+
+        point1 = SpherePoint(afwGeom.Point3D(0.1, 0.1, 0.1))
+        point2 = SpherePoint(afwGeom.Point3D(0.6, 0.6, 0.6))
+        sep1 = refPoint.separation(point1)
+        sep2 = refPoint.separation(point2)
+        sepTrue = 54.735610317245339*degrees
+
+        self.assertAnglesNearlyEqual(sepTrue, sep1)
+        self.assertAnglesNearlyEqual(sepTrue, sep2)
+
+    def testAtPoleValue(self):
+        """Test if atPole() returns the expected value.
+        """
+        poleList = [
+            SpherePoint(0.0*degrees, -90.0*degrees),
+            SpherePoint(42.0*degrees, 324000.0*afwGeom.arcseconds),
+            SpherePoint(-386.0*degrees, afwGeom.HALFPI*radians),
+            SpherePoint(afwGeom.Point3D(0.0, 0.0, 1.0)),
+            SpherePoint(afwGeom.Point3D(0.0, 0.0, -1.0)),
+        ]
+        nonPoleList = [
+            SpherePoint(0.0*degrees, (-90.0 + 1e-14)*degrees),
+            SpherePoint(0.0*degrees, 323999.0*afwGeom.arcseconds),
+            SpherePoint(afwGeom.Point3D(9.9e-7, 0.0, 1.0)),
+            SpherePoint(0.0*degrees, nan*degrees),
+        ]
+
+        for pole in poleList:
+            self.assertIsInstance(pole.atPole(), bool)
+            self.assertTrue(pole.atPole())
+
+        for nonPole in nonPoleList:
+            self.assertIsInstance(nonPole.atPole(), bool)
+            self.assertFalse(nonPole.atPole())
+
+    def testIsFiniteValue(self):
+        """Test if isFinite() returns the expected value.
+        """
+        finiteList = [
+            SpherePoint(0.0*degrees, -90.0*degrees),
+            SpherePoint(afwGeom.Point3D(0.1, 0.2, 0.3)),
+        ]
+        nonFiniteList = [
+            SpherePoint(0.0*degrees, nan*degrees),
+            SpherePoint(nan*degrees, 0.0*degrees),
+            SpherePoint(inf*degrees, 0.0*degrees),
+            SpherePoint(-inf*degrees, 0.0*degrees),
+            SpherePoint(afwGeom.Point3D(nan, 0.2, 0.3)),
+            SpherePoint(afwGeom.Point3D(0.1, inf, 0.3)),
+            SpherePoint(afwGeom.Point3D(0.1, 0.2, -inf)),
+        ]
+
+        for finite in finiteList:
+            self.assertIsInstance(finite.isFinite(), bool)
+            self.assertTrue(finite.isFinite())
+
+        for nonFinite in nonFiniteList:
+            self.assertIsInstance(nonFinite.isFinite(), bool)
+            self.assertFalse(nonFinite.isFinite())
+
+    def testGetItemError(self):
+        """Test if indexing correctly handles invalid input.
+        """
+        point = SpherePoint(afwGeom.Point3D(1.0, 1.0, 1.0))
+
+        with self.assertRaises(pexEx.OutOfRangeError):
+            point[2]
+        with self.assertRaises((pexEx.OutOfRangeError, OverflowError)):
```

Comment:
I am a bit disappointed this fails. In Python I would expect point[-1] to return latitude (last element) and point[-2] to return longitude (2nd-to-last element).

I encourage you to spend a little time trying to fix this. For instance it may suffice to hide the C++ operator[] from SWIG (using %ignore, for instance). The SWIG wrapper already has `__iter__` and `__len__` and that might suffice. If not, defining `__getitem__` should definitely do it.

If you do fix this, then indices in range [-2, 1] are all valid, and any other might throw a standard Python exception instead of an LSST C++ exception (but it depends how you implement it).